js/utility.js

```diff
@@ -22,4 +22,54 @@ export function unhideMultipleElements(array) {
     for (let i = 0; i < array.length; i++) {
         unhideElement(array[i]);
     }
-}
\ No newline at end of file
+}
+
+const levelPopUp = document.querySelector(".levelPopUp");
+
+function triggerPopUp(startWidth, endWidth, element) {
+    const duration = 300; // ms
+    const startTime = performance.now();
+
+    function animate(currentTime) {
+        const elapsed = currentTime - startTime;
+        const progress = Math.min(elapsed / duration, 1);
+
+        const currentWidth = startWidth + (endWidth - startWidth) * progress;
+        element.style.clipPath = `rect(0px ${currentWidth}% 100% 0px round 0%)`
+
+        if (progress < 1) {
+            requestAnimationFrame(animate);
+        }
+    }
+    requestAnimationFrame(animate);
+}
+
+function clearPopUp(elementClass) {
```

Comment:
Following up from my other comment
```suggestion
function clearPopUp(elementClass) {
    triggerPopUp(100, 0);

    setTimeout(() => {
        hideElement(element);
        levelPopUp.classList.remove(elementClass);
    }, 400);
}
```